packages/workers-shared/router-worker/tests/index.test.ts

```diff
@@ -0,0 +1,66 @@
+import { createExecutionContext } from "cloudflare:test";
+import { describe, expect, it } from "vitest";
+import { default as worker } from "../src/index";
+
+describe("unit tests", async () => {
+	it("fails if specify running user worker ahead of assets, without user worker", async () => {
+		const request = new Request("https://example.com");
+		const ctx = createExecutionContext();
+
+		const env = {
+			CONFIG: {
+				invoke_user_worker_ahead_of_assets: true,
+				has_user_worker: false,
+			},
+		} as typeof env;
+
+		void expect(
+			async () => await worker.fetch(request, env, ctx)
+		).rejects.toThrowError(
+			"Fetch for user worker without having a user worker binding"
+		);
+	});
+
+	it("it returns fetch from user worker when invoke_user_worker_ahead_of_assets true", async () => {
+		const request = new Request("https://example.com");
+		const ctx = createExecutionContext();
+
+		const env = {
+			CONFIG: {
+				invoke_user_worker_ahead_of_assets: true,
+				has_user_worker: true,
+			},
+			USER_WORKER: {
+				async fetch(_: Request): Promise<Response> {
+					return new Response("hello from user worker");
+				},
+			},
```

Comment:
Could you add an ASSET_WORKER in here too, just to prove that having an ASSET_WORKER does not force the request to go there rather than the USER_WORKER.